common/workunit/workunit.cpp

```diff
@@ -2446,7 +2426,11 @@ class CDaliWorkUnitFactory : public CWorkUnitFactory, implements IDaliClientShut
     virtual void deleteRepository(bool recreate)
     {
         Owned<IRemoteConnection> conn = sdsManager->connect("/WorkUnits", session, RTM_LOCK_WRITE, SDS_LOCK_TIMEOUT);
-        conn->close(true);
+        if (conn)
+            conn->close(true);
+        conn.setown(sdsManager->connect("/GraphProgress", session, RTM_LOCK_WRITE, SDS_LOCK_TIMEOUT));
+        if (conn)
+            conn->close(true);
     }
     virtual void createRepository()
```

Comment:
There may be assumptions that "/WorkUnits" exists, it is created (if missing) when Dali starts.
Should createRepository do the reverse of deleteRepository and create /WorkUnits and /GraphProgress?

I suspect anywhere that assumes either exist, should be fixed to create or it or handle it being missing instead.